### ISSUE TITLE
Using name (from window.name) should be an offense

### ISSUE THREAD

**Issue by Dorian** [eslint/eslint#8572]
```js
if (typeof name == "string") { console.log("oops") }
```

`name` comes from `window.name`, so it should be referenced as `window.name`.

[x] Enforces code style
[x] Warns about a potential error
[x] Suggests an alternate way of doing something

It's too easy to assume `name` is not defined but it is on window (e.g. https://developer.mozilla.org/en-US/docs/Web/API/Window/name)

Either using `window.name`, `global.name`, etc. would be fine.

I think it's such a common and unintuitive bug that it should be the default.

Source: https://twitter.com/_jayphelps/status/862100526703403008

**Comment by not-an-aardvark** [comment]
Does [no-restricted-globals](http://eslint.org/docs/rules/no-restricted-globals) handle this use case?

**Comment by Dorian** [comment]
@not-an-aardvark Why not make it a default?

**Comment by not-an-aardvark** [comment]
Do you mean a default for the `no-restricted-globals` rule, or a default for ESLint in general? By design, no ESLint rules are on by default.

**Comment by eslintbot** [comment]
Hi @Dorian, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by Dorian** [comment]
@not-an-aardvark Why is that? Why not have a default configuration? (Sorry, I'm biased because I use rubocop a lot)

Then I could just `npm install -g eslint` and `eslint index.js`

<img width="300" alt="screen shot 2017-05-11 at 10 15 17" src="https://cloud.githubusercontent.com/assets/129149/25962461/c52a81e2-3632-11e7-9fb4-6aa8209f8952.png">

<img width="300" alt="screen shot 2017-05-11 at 10 15 01" src="https://cloud.githubusercontent.com/assets/129149/25962462/c53c6182-3632-11e7-985b-ec7c52ecd381.png">

<img width="300" alt="screen shot 2017-05-11 at 10 16 11" src="https://cloud.githubusercontent.com/assets/129149/25962510/e1f7c5fa-3632-11e7-8465-f84c3bc875a9.png">

**Comment by platinumazure** [comment]
@Dorian The simple answer is that your idea of a sensible set of default restricted globals will not be the same as others', and so ESLint tries to be unopinionated by not imposing a set of defaults that some people will disagree with. Instead, `no-restricted-globals` is designed to be flexible and to work on a case-by-case basis, so if you want to forbid certain globals, you just throw them into your configuration and nobody else has to worry about it.

If you have difficulty setting up your configuration to block the `name` global, please feel free to stop by our [Gitter chat](https://gitter.im/eslint/eslint) and we'll be happy to help.

If you want to enhance our configuration initializer feature to allow users to specify restricted globals, properties, or whatever else, we could discuss that possible core enhancement (although I personally would think the maintenance headache would not be worth the extra functionality). Feel free to create a new issue for that if you like.

**Comment by Dorian** [comment]
> ESLint tries to be unopinionated by not imposing a set of defaults that some people will disagree with

@platinumazure Ok, I didn't know that was a strong opinion of this project. I believe more in ["convention over configuration"](http://rubyonrails.org/doctrine/#convention-over-configuration)

**Comment by not-an-aardvark** [comment]
@Dorian At one point, ESLint had a set of rules that was on by default. However, this was changed later on -- see https://github.com/eslint/eslint/issues/2100 for the discussion and reasoning that led to that decision.

Among other things, one disadvantage of having rules on by default is that ESLint would have to make a lot of code style decisions for the user. There are a huge number of common code styles, so the majority of users would disagree with some of ESLint's decisions and have to explicitly disable or reconfigure rules anyway. This would defeat the point of having everything work with no configuration.

**Comment by Dorian** [comment]
@not-an-aardvark Sure, the thing is `npm install -g eslint` and `eslint index.js` literally doesn't work, is it cool if I fix this specific use-case?

**Comment by not-an-aardvark** [comment]
That is working as designed, so I don't understand what fix you're proposing.

If you want to enforce a common styleguide without manually configuring a bunch of rules yourself, you can use a [sharable config](http://eslint.org/docs/user-guide/configuring#using-a-shareable-configuration-package) such as [eslint-config-airbnb](https://github.com/airbnb/javascript).

**Comment by platinumazure** [comment]
@Dorian You can ensure `name` is flagged by setting configuration:

```js
{
    "rules": {
        "no-restricted-globals": ["error", "name"]
    }
}
```

We don't guarantee that `npm install -g` will automatically configure everything exactly as users want it-- if we could, then we could have default rule settings, but as noted above, that is impossible. As @not-an-aardvark notes above, you can also use a shareable config if you are okay with following that config's style guide.

**Comment by not-an-aardvark** [comment]
Closing because this is working as intended.
